lib/pact/reification.rb

```diff
@@ -14,7 +14,7 @@ module Reification
     def self.from_term(term)
       case term
       when Pact::Term, Regexp, Pact::SomethingLike, Pact::ArrayLike
-        from_term(term.generate)
+        from_term(Expgen.gen(term))
```

Comment:
You'll need to split out the `Regexp` from the other classes, and only call `Expgen.gen(term)` for Regexp. The other classes still need to use `term.generate`.